programs/address-lookup-table/src/state.rs

```diff
@@ -133,6 +134,39 @@ impl<'a> AddressLookupTable<'a> {
         Ok(())
     }
 
+    /// Lookup addresses for provided table indexes. Since lookups are performed on
+    /// tables which are not read-locked, this implementation needs to be careful
+    /// about resolving addresses consistently.
+    pub fn lookup(
+        &self,
+        current_slot: Slot,
+        indexes: &[u8],
+        slot_hashes: &SlotHashes,
+    ) -> Result<Vec<Pubkey>, AddressLookupError> {
+        if !self.meta.is_active(current_slot, slot_hashes) {
+            // Once a lookup table is no longer active, it can be closed
+            // at any point, so returning a specific error for deactivated
+            // lookup tables could result in a race condition.
+            return Err(AddressLookupError::LookupTableAccountNotFound);
+        }
+
+        // If the address table was extended in the same slot in which it is used
+        // to lookup addresses for another transaction, the recently extended
+        // addresses are not considered active and won't be accessible.
+        let active_addresses_len = if current_slot > self.meta.last_extended_slot {
+            self.addresses.len()
+        } else {
+            self.meta.last_extended_slot_start_index as usize
+        };
```

Comment:
nit: how about factoring this into a separate function e.g. `get_active_addresses_len()`?

Reply:
what value would that add? I think it's fine there

Reply:
It's mainly for some extra code-as-documentation, and because it's encapsulating an important part of address lookup tables, which is that you should only look until `last_extended_slot_start_index` during the extension slot.  Even if that's never exposed outside, it can save you later.  But either way, it's just a nit

Reply:
That's a good reason, updated!